odoo/models.py

```diff
@@ -3632,6 +3632,21 @@ def get_base_url(self):
             raise ValueError("Expected singleton or no record: %s" % self)
         return self.env['ir.config_parameter'].sudo().get_param('web.base.url')
 
+    def _parse_companies(self, companies):
+        if companies is None:
+            return self.env.company.ids
+        elif isinstance(companies, BaseModel):
+            return companies.ids
+        elif isinstance(companies, (list, tuple)):
+            return companies
+        return [companies]
+
+    def _check_company_domain_parent_of(self, companies=None):
+        return ['|', ('company_id', '=', False), ('company_id', 'parent_of', self._parse_companies(companies))]
+
+    def check_company_domain(self, companies=None):
```

Comment:
Please make the method private, as making it public has no purpose.
And add a docstring to document what it should be used for.
Also, I recommend to make the argument mandatory.  The parameter should be explicit IMHO.

Reply:
It could be needed by the web client to build a domain correctly if needed

Reply:
A public method requires a different API (arguments cannot be records).
If you don't have a concrete case, it does not need to be public.

Reply:
The argument can be anything, including a list of ids